cmd/ptp4u/main.go

```diff
@@ -114,9 +116,13 @@ func main() {
 	st := stats.NewJSONStats()
 	go st.Start(c.MonitoringPort)
 
+	// drain check
+	check := &drain.FileDrain{FileName: "/var/tmp/kill_ptp4u"}
+
 	s := server.Server{
 		Config: c,
 		Stats:  st,
+		Check:  check,
```

Comment:
So if you make Check an array of interfaces you can stick array of different checks. Everyone can use the public “file” and add as many additional as they want without reimplementing “file” check internally.